app/src/main/java/org/sopt/and/ui/main/MainViewModel.kt

```diff
@@ -0,0 +1,23 @@
+package org.sopt.and.ui.main
+
+import androidx.lifecycle.ViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.update
+
+class MainViewModel : ViewModel() {
+    private val _bottomBarVisible = MutableStateFlow(BarStatus())
+    val bottomBarVisible = _bottomBarVisible.asStateFlow()
+
+    fun setBarVisible(isVisible: Boolean) {
+        _bottomBarVisible.update {
+            it.copy(
+                bottomBarVisible = isVisible
+            )
+        }
+    }
+}
+
+data class BarStatus(
+    val bottomBarVisible: Boolean = false
+)
```

Comment:
단순 Boolean 값만을 다룬다면 BarStatus 클래스 선언 없이 이렇게도 사용해도 괜찮을거 같습니다!
```suggestion
class MainViewModel : ViewModel() {
    private val _bottomBarVisible = MutableStateFlow(false)
    val bottomBarVisible = _bottomBarVisible.asStateFlow()

    fun setBarVisible(isVisible: Boolean) {
        _bottomBarVisible.value = isVisible
    }
}
```

Reply:
확인했습니다 과하지않겠습니다